src/Core/Result.php

```diff
@@ -85,20 +85,22 @@ public function __destruct()
      */
     final public function resolve(?float $timeout = null): bool
     {
-        if (!isset($this->response)) {
+        if ($this->isResolved) {
             return true;
         }
 
-        try {
-            if (null !== $timeout && $this->httpClient->stream($this->response, $timeout)->current()->isTimeout()) {
-                return false;
+        foreach ($this->httpClient->stream($this->response, $timeout) as $chunk) {
+            try {
+                if ($chunk->isTimeout()) {
+                    return false;
+                }
+                if ($chunk->isFirst()) {
+                    $statusCode = $this->response->getStatusCode();
+                    break;
+                }
+            } catch (TransportExceptionInterface $e) {
+                throw new NetworkException('Could not contact remote server.', 0, $e);
             }
-            $statusCode = $this->response->getStatusCode();
-        } catch (TransportExceptionInterface $e) {
-            // When a network error occurs
-            $this->isResolved = true;
```

Comment:
revert

Reply:
this means the method becomes order-sentitive

Reply:
refactored